lib/exercism/github_source.rb

```diff
@@ -0,0 +1,25 @@
+class GithubSource
+  attr_reader :github, :user, :slug, :code
+
+  def initialize(submission)
+    @github = Github.new
+    @user = submission.user
+    @slug = submission.slug
+    @code = submission.code
+  end
+
+  def solution
+    trees = github.git_data.trees.get user.username, slug, code
+
+    res = {}
+    trees.tree.each do |tree_item|
+      if tree_item.type == 'blob'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.